MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -1619,12 +1623,31 @@ private void updateLocation(@NonNull LocationUpdate locationUpdate, boolean from
     }
     CameraPosition currentCameraPosition = mapboxMap.getCameraPosition();
     boolean isGpsNorth = getCameraMode() == CameraMode.TRACKING_GPS_NORTH;
-    locationAnimatorCoordinator.feedNewLocation(
-      getTargetLocationWithIntermediates(locationUpdate.getLocation(), locationUpdate.getIntermediatePoints()),
-      currentCameraPosition,
-      isGpsNorth,
-      fromLastLocation ? Long.valueOf(0L) : locationUpdate.getAnimationDuration());
-    updateAccuracyRadius(locationUpdate.getLocation(), false);
+
+
+    //Performance improvement: Check if new location requires a redraw to be performed,
+    //taking the pixel tolerance into account.
+    //By default, the tolerance is 1 pixel, meaning that the new location must impact at least
+    //a different pixel on the screen to justify a redraw.
+    boolean updateNeeded = (lastLocation == null);
+    if (updateNeeded == false) {
+      PointF lastPoint = this.mapboxMap.getProjection().toScreenLocation(
+              new LatLng(lastLocation.getLatitude(), lastLocation.getLongitude()));
+      PointF updatedPoint = this.mapboxMap.getProjection().toScreenLocation(
+              new LatLng(locationUpdate.getLocation().getLatitude(), locationUpdate.getLocation().getLongitude()));
+
+      updateNeeded = ((Math.abs(updatedPoint.x - lastPoint.x) >= options.locationTolerance())
+                      || (Math.abs(updatedPoint.y - lastPoint.y) >= options.locationTolerance()) );
+    }
+
+    if (updateNeeded ) {
+      locationAnimatorCoordinator.feedNewLocation(
+              getTargetLocationWithIntermediates(locationUpdate.getLocation(), locationUpdate.getIntermediatePoints()),
+              currentCameraPosition,
+              isGpsNorth,
+              fromLastLocation ? Long.valueOf(0L) : locationUpdate.getAnimationDuration());
+    }
+    updateAccuracyRadius(locationUpdate.getLocation(), false, false);
     lastLocation = locationUpdate.getLocation();
```

Comment:
This assignment has to happen inside of the if clause when an update is actually dispatched. Otherwise, if the tolerance is big enough, we could starve the animator and never reach the threshold while losing more and more precision resulting in either the location indicator not moving at all or jumping in random intervals.

Reply:
Yes, and this assignment can also be done after the radius display as the accuracy field should be taken into consideration.
I updated the code to make it each time necessary.

Reply:
Sorry, you're totally right, the previous radius is stored, i'll fix that right now.